docs/src/languages/java/migration.md

```diff
@@ -0,0 +1,166 @@
+---
+title: "Moving a Java application to Platform.sh"
+weight: 2
+sidebarTitle: "Moving to Platform.sh"
+---
+It is common to have a Java application that wants to migrate to Platform.sh. It has support to several styles of Java application such as monolith, microservices, stateful, and stateless.
+## Minimum Requirement
+
+To run a Java application at Platform.sh you must check: 
+
+* [A supported Java version](https://docs.platform.sh/languages/java.html#supported-versions)
+* [A build management](https://docs.platform.sh/languages/java.html#support-build-automation)
+  * [Gradle](https://docs.gradle.org/current/userguide/gradle_wrapper.html)
+  * [Maven](https://maven.apache.org/) 
+  * [Maven Wrapper](https://www.baeldung.com/maven-wrapper)
+  * [Ant](https://ant.apache.org/)
+* A Git Repository:
+  * [GitHub](https://docs.platform.sh/integrations/source/github.html)
+  * [BitBucket](https://docs.platform.sh/integrations/source/bitbucket.html)
+  * [GitLab](https://docs.platform.sh/integrations/source/gitlab.html)
+  * The default Git repository provided by Platform.sh
+
+{{< note >}}
+A container application cannot be bigger than **8 GB**, see  [tunning](https://docs.platform.sh/languages/java/tuning.html) to more details.
+{{< /note >}}
+
+## Monolith/Single Application
+
+To start a Java application, you need to understand the [Platform.sh structure](https://docs.platform.sh/overview/structure.html).  At minimum you will need three [YAML files](https://docs.platform.sh/configuration/yaml.html): 
+
+1. [Application](https://docs.platform.sh/configuration/app.html)
+2. [Route](https://docs.platform.sh/configuration/routes.html)
+
+### Application
+
+```yaml
+name: app
+type: "java:<version>"
+disk: 1024
+hooks:
+    build: 2
+web:
+    commands:
+        start: 3
+```
+
+1. [A Java version](https://docs.platform.sh/languages/java.html#supported-versions), e,g.: `java:11`
```

Comment:
Oh!  I see, you were using the numbers as footnotes.  That makes the removal of the "1" I suggested not make sense. :smile:  I'm not sure that footnoting a YAML snippet works, though.  Maybe if the numbers used "[2]" instead of just "2"?

I'm not sure what works well here.  @chadwcarlson?